### ISSUE TITLE
Consider moving to yt-dlp

### ISSUE THREAD

**Issue by TheFrenchGhosty** [TheFrenchGhosty/TheFrenchGhostys-YouTube-DL-Archivist-Scripts#37]
Youtube-dlc has been abandonned, see https://github.com/blackjack4494/yt-dlc/issues/296

A fork of it has been released named yt-dlp: https://github.com/pukkandan/yt-dlp

Consider moving to it.

**Comment by vxbinaca** [comment]
why?

**Comment by TheFrenchGhosty** [comment]
@vxbinaca "Youtube-dlc has been abandonned, see blackjack4494/yt-dlc#296"

**Comment by vxbinaca** [comment]
I switched back to YouTube-dl. Is there some extractor functionality you feel is better in this fork?

**Comment by TheFrenchGhosty** [comment]
Comments.

And the developers aren't assholes.

**Comment by vxbinaca** [comment]
Lemme give it a spin this week and if it works with the extractors I want I can switch. YTDL is lagging behind again.

**Comment by Mennaruuk** [comment]
Hi! I think you should consider [**haruhi-dl**](https://haruhi.download)! It's a libre youtube-dl fork maintained by three wonderful people, and they work really hard (and fast) to fix and even create new extractors for any site! They've already implemented [tons of extractors](https://git.sakamoto.pl/laudompat/haruhi-dl/-/tree/master/haruhi_dl/extractor), and some sites that don't work with youtube-dl work with haruhi-dl (looking at you, [Crunchyroll](https://git.sakamoto.pl/laudompat/haruhi-dl/-/blob/master/haruhi_dl/extractor/crunchyroll.py)! [Funkwhale](https://git.sakamoto.pl/laudompat/haruhi-dl/-/blob/master/haruhi_dl/extractor/funkwhale.py) is another). Also, they're planning to generate binary builds as well as implement tools like [Playwright](https://github.com/microsoft/playwright/blob/master/README.md), which allows for native browser automation on Chromium, WebKit, and Firefox. So there are a lot of exciting things coming up for this amazing fork, and I think it'd be a great choice for this repo.

**Comment by vxbinaca** [comment]
I'm not implementing a downloader that does Crunchyroll or Netflix thats a recipe for disaster. Tubeup is our project not AltCensoreds. I have to work with the internet archive not against them making trouble for them.

**Comment by Mennaruuk** [comment]
I understand. In this case, [yt-dlp](https://github.com/pukkandan/yt-dlp) would be a great replacement. The author pukkandan is constantly making new changes to improve it (check [changelog](https://github.com/pukkandan/yt-dlp/blob/master/Changelog.md)). I can see in his latest update they added date/time formatting in output template, that can be of use for archiving. And they've been helping complete feature requests such as playlist and channel metadata download. An overall wonderful fork.

**Comment by TheFrenchGhosty** [comment]
Done in: https://github.com/TheFrenchGhosty/TheFrenchGhostys-YouTube-DL-Archivist-Scripts/commit/3d826c132b9c7a5539255c8cbcc91f428879c375 and https://github.com/TheFrenchGhosty/TheFrenchGhostys-YouTube-DL-Archivist-Scripts/commit/6d9a3a356540ec3317ff822304d1b783fb3aa27b

**Comment by selfisekai** [comment]
@vxbinaca hi, hdl maintainer here. Crunchyroll extractor is actually implemented in ytdl since 2013, as says the commit history, and the only change we made [so far](https://git.sakamoto.pl/laudompat/haruhi-dl/-/commits/master/haruhi_dl/extractor/crunchyroll.py) to it was adding the letter "s" to avoid a one HTTP redirect. We're also not even touching Netflix and any DRM-protected content. Most of the extractor changes we made were about social media, news pages and YouTube.
